src/components/member/PasswordSetting.js

```diff
@@ -0,0 +1,228 @@
+    justifyContent: 'flex-start',
+    alignItems: 'center',
+    margin: '12px 0px',
+  },
+  buttonItem: {
+    display: 'flex',
+    padding: '12px',
+    flexDirection: 'column',
+    justifyContent: 'flex-end',
+    alignItems: 'flex-start',
+  },
+  note: {
+    fontSize: '12px',
+    margin: '4px 0 2px',
+  },
+}));
+
+const PasswordSetting = ({ user }) => {
+  const { t } = useTranslation();
+  const classes = useStyles();
+
+  const [currentPassword, setCurrentPassword] = useState('');
+  const [newPassword, setNewPassword] = useState('');
+  const [confirmPassword, setConfirmPassword] = useState('');
+  const [currentPasswordError, setCurrentPasswordError] = useState(null);
+  const [newPasswordError, setNewPasswordError] = useState(null);
+  const [confirmPasswordError, setConfirmPasswordError] = useState(null);
+  const { mutate: onUpdatePassword } = useMutation(
+    MUTATION_KEYS.UPDATE_PASSWORD,
+  );
+
+  const userId = user.get('id');
+
+  const onClose = () => {
+    setCurrentPassword('');
+    setNewPassword('');
+    setConfirmPassword('');
+  };
+
+  const handleChangePassword = () => {
+    const checkingCurrentPassword = passwordValidator(currentPassword);
+    const checkingNewPassword = passwordValidator(newPassword);
+    const checkingConfirmPassword = passwordValidator(confirmPassword);
+    // verify empty fields
+    if (
+      user.get('password') &&
+      (checkingCurrentPassword ||
+        checkingNewPassword ||
+        checkingConfirmPassword)
+    ) {
+      setCurrentPasswordError(checkingCurrentPassword);
+      setNewPasswordError(checkingNewPassword);
+      setConfirmPasswordError(checkingConfirmPassword);
+    } else {
+      const checkingPasswordUpdate = newPasswordValidator(
+        currentPassword,
+        newPassword,
+        confirmPassword,
+      );
+      if (checkingPasswordUpdate) {
+        toast.error(checkingPasswordUpdate);
+      } else {
+        const checkingStrength = strengthValidator(newPassword);
+        if (checkingStrength) {
+          toast.error(checkingStrength);
+        } else {
+          onUpdatePassword({
+            id: userId,
+            password: newPassword,
+            currentPassword,
+          });
+          onClose();
+        }
+      }
+    }
+  };
+
+  const handleCurrentPasswordInput = (event) => {
+    setCurrentPassword(event.target.value);
+    setCurrentPasswordError(passwordValidator(event.target.value));
+  };
+  const handleNewPasswordInput = (event) => {
+    setNewPassword(event.target.value);
+    setNewPasswordError(passwordValidator(event.target.value));
+  };
+  const handleConfirmPasswordInput = (event) => {
+    setConfirmPassword(event.target.value);
+    setConfirmPasswordError(passwordValidator(event.target.value));
+  };
+
+  return (
+    <>
+      <Grid container spacing={3} className={classes.mainContainer}>
+        <Grid item xs={8}>
+          <Grid item xs={12}>
+            {user.get('password') !== null ? (
```

Comment:
`user.get('password')` ? Your password could be empty (usually not but maybe)
And since only the text changed, maybe you do the check in `Typography`?

Reply:
Left only the 'Change password' text as we are not going to evaluate the password from the user

Reply:
user.get('password') returns True if the user has a password set or false if not. 
I refactored this check inside `Typography ` element so that it appears based on that.